### PR TITLE
Add function def clauses

### PR DESCRIPTION
**Pull request by jackh726**
Missed in https://github.com/rust-lang/chalk/pull/449

Also Sized test for `Never`

### CODE REVIEW

File: book/src/clauses/well_known_traits.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Some common examples of auto traits are `Send` and `Sync`.
 | str             | 📚    | 📚    | ⚬     | ⚬     | ⚬   |  ⚬  |  ⚬     |  ⚬       |    ❌        |
 | never type      | 📚   |  📚   |  ✅   |  ⚬    |  ⚬    | ⚬   |   ⚬    |  ⚬       |   ❌       |
 | trait objects   | ⚬    | ⚬    | ⚬     |  ✅    | ⚬    | ⚬   | ⚬      |  ⚬       |    ⚬        |
+| functions defs  | ✅    | ✅    | ✅     | ⚬     | ⚬    |  ⚬  | ⚬      |  ⚬       |    ❌         |
```
Comment (jackh726) [inline book/src/clauses/well_known_traits.md:39]: I actually don't know, should `FnDef`s impl `Fn`?

Comment (flodiebold) [reply book/src/clauses/well_known_traits.md:39]: [Yes](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c91518443d32a200da4f0a299bfe9b44).

File: tests/test/fn_def.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "Unique"
+        }
+    }
+}
+
+#[test]
+fn fn_def_is_copy() {
+    test! {
+        program {
+            #[lang(sized)]
```
Comment (bjorn3) [inline tests/test/fn_def.rs:38]: This should be `copy`, not `sized`.

File: tests/test/fn_def.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            "Unique"
+        }
+    }
+}
+
+#[test]
+fn fn_def_is_clone() {
+    test! {
+        program {
+            #[lang(sized)]
```
Comment (bjorn3) [inline tests/test/fn_def.rs:55]: This should be `clone`, not `sized`.

**Comment by jackh726** [comment]
Well shoot...I'll just include these in the chalk-base PR.

File: tests/test/fn_def.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
+use super::*;
```
Comment (Mcat12) [inline tests/test/fn_def.rs:1]: Another boo-boo, these tests aren't run because the module is not listed in `tests/test/mod.rs`

Comment (Mcat12) [reply tests/test/fn_def.rs:1]: They fail in the parser because it expects `;` instead of `{ }`.

Comment (jackh726) [reply tests/test/fn_def.rs:1]: 🙄

**Pull request merged**